src/firmware/shared/circular_buffer_test.cpp

```diff
@@ -0,0 +1,130 @@
+extern "C"
+{
+#include "firmware/shared/circular_buffer.h"
+
+#include "shared/constants.h"
+}
+
+#include <gtest/gtest.h>
+
+class CircularBufferTest : public testing::Test
+{
+   protected:
+    virtual void SetUp(void)
+    {
+        circular_buffer = circular_buffer_create(10);
+    }
+
+    CircularBuffer_t* circular_buffer;
+};
+
+TEST_F(CircularBufferTest, circular_buffer_exact_full)
+{
+    float mockData[]     = {1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0};
```

Comment:
variables should be `snake_case` according to our convention (just leaving 1 comment but I noticed this in at least this file and the buffer .c file)

Reply:
Fixed. For `circular_buffer.c`, I changed `isFull` to `is_full`. Do you think the function name should change to `circular_buffer_is_full()` or remain `circular_buffer_isFull`?

Reply:
`circular_buffer_isFull` is the correct way. Variables and function names follow different conventions ([although it's a bit weirder in C because of the layer naming](https://github.com/UBC-Thunderbots/Software/blob/master/docs/code-style-guide.md#c-specific-guidelines))